dotcom-rendering/src/lib/byline.ts

```diff
@@ -1,11 +1,38 @@
-export const getContributorTags = (tags: TagType[]): TagType[] => {
-	return tags.filter((t) => t.type === 'Contributor');
+interface ContributorTag extends TagType {
+	type: 'Contributor';
+}
+
+const isContributor = (tag: TagType): tag is ContributorTag =>
+	tag.type === 'Contributor';
+
+/**
+ * Get the sole contributor of an article.
+ * This only works if all of the following are true:
```

Comment:
'This only works' implies to me that it might break if some of the conditions don't hold. But my understanding is that it won't break, and that it's actually the desired behaviour that it will only return a contributor in the case where all the conditions hold.

If this is correct, maybe this comment could be changed to something like `An article has a sole contributor only if all of the following are true:`?

Reply:
And given that a lot of the rendering decisions which invoke `getSoleContributor` are really relying on both of these conditions, I wonder if it's worth spelling out in a bit more detail here what it is that makes these conditions relevant to those rendering decisions? (i.e. why we don't show any images if there is more than one contributor, and why it is that byline and tag title can diverge?)